project/profiles/capacity4more/modules/c4m/wysiwyg/c4m_wysiwyg_ckeditor/c4m_wysiwyg_ckeditor.module

```diff
@@ -104,3 +104,116 @@ function c4m_wysiwyg_ckeditor_media_wysiwyg_format_form_prepare_alter(
       'c4m_wysiwyg_ckeditor'
     ) . '/c4m_wysiwyg_ckeditor.js';
 }
+
+/**
+ * Define all external domains that use cookies.
+ *
+ * @return array
+ */
+function _c4m_wysiwyg_ckeditor_ckeditor_iframes_with_cookies() {
+  return [
+    'youtube',
+    'archive',
+    'dailymotion',
+    'google',
+    'slideshare',
+    'vimeo',
+    'ustream',
+    'prezi',
+    'issuu',
+    'scribd',
+    'tubechop',
+    'twitter',
+    'twitframe',
+    'mapbox',
+    'soundcloud',
+    'audioboom',
+    'mixcloud',
+    'mindmeister'
+  ];
+}
+
+/**
+ * Implements hook_filter_info().
+ */
+function c4m_wysiwyg_ckeditor_filter_info() {
+  $filters['c4m_wysiwyg_ckeditor_iframe_cck'] = array(
+    'title' => t('Iframe Cookie Consent Filter'),
+    'description' => t('Checks iframes and adds the EU Cookie Consent kit to it if applicable'),
+    'process callback' => '_c4m_wysiwyg_ckeditor_iframe_cck_filter_process',
+    'tips callback' => '_c4m_wysiwyg_ckeditor_iframe_cck_filter_tips',
+  );
+
+  return $filters;
+}
+
+/**
+ * Returns tips for defined filter.
+ *
+ * @param $filter
+ * @param $format
+ * @param bool $long
+ * @return string
+ */
+function _c4m_wysiwyg_ckeditor_iframe_cck_filter_tips(
+  $filter,
+  $format,
+  $long = FALSE
+) {
+  return t('Checks the source input for iframes of a specific service and adds 
+  the EU Cookie Consent kit to it if applicable.');
+}
+
+/**
+ * Prepares markup for external iframes.
+ */
+function _c4m_wysiwyg_ckeditor_iframe_cck_filter_process(
+  $text,
+  $filter,
+  $format,
+  $langcode,
+  $cache,
+  $cache_id
+) {
+  // Check if there is an iframe in the text.
+  if (!preg_match('/<\s*iframe /mis', $text)) {
+    return;
+  }
+  // Load the text into a dom object.
+  $dom = new DOMDocument();
+  $dom->loadHTML($text);
+  $services = _c4m_wysiwyg_ckeditor_ckeditor_iframes_with_cookies();
+  $services_replaced = [];
+
+  foreach ($dom->getElementsByTagName('iframe') as $iframe) {
+    if (!$iframe->hasAttribute('src')) {
+      break;
+    }
+    $src = $iframe->getAttribute('src');
+    $host = parse_url($src, PHP_URL_HOST);
+    // Get the domain name.
+    preg_match('/([^.]+)\.[^.]+$/', $host, $matches);
+    $service = $matches[1];
+    // Alter media source so it will be rendered after the cookie policy is
+    // accepted.
+    $iframe->setAttribute(
```

Comment:
Src only needs to be changed if the service is in our array, so please move this within the IF statement.